common/coin_support/abi_decode.h

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file    abi_decode.h
+ * @author  Cypherock X1 Team
+ * @brief   Title of the file.
+ *          Short description of the file
+ * @copyright Copyright (c) 2022 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/" target=_blank>https://mitcc.org/</a>
+ * 
+ */
+#ifndef ABI_DECODE
+#define ABI_DECODE
+
+#include <assert.h>
+#include <stdio.h>
+#include "eth.h"
+
+/* MACROS
+ *****************************************************************************/
+#define ABI_ELEMENT_SZ_IN_BYTES						(32)
+#define Abi_uint256_e_SZ_IN_BYTES					(32)
+
+#define Abi_address_e_SZ_IN_BYTES					(20)
+#define Abi_address_e_OFFSET_BE	(Abi_uint256_e_SZ_IN_BYTES - Abi_address_e_SZ_IN_BYTES)
+
+#define ABI_DYN_METADATA_SZ_IN_BYTES				(4)

```

Comment:
let's mention purpose of this. As in, if we are trying/expecting a practical upper limit or something similar.

Reply:
ok added a comment